core/src/main/java/io/questdb/griffin/engine/functions/date/NowFunctionFactory.java

```diff
@@ -48,25 +47,24 @@ public boolean isRuntimeConstant() {
 
     @Override
     public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) {
-        return new Func();
+        return new Func(sqlExecutionContext);
     }
 
     private static class Func extends TimestampFunction implements Function {
-        private SqlExecutionContext context;
+        private final SqlExecutionContext context;
 
-        @Override
-        public long getTimestamp(Record rec) {
-            return context.getNow();
+        private Func(SqlExecutionContext sqlExecutionContext) {
+            context = sqlExecutionContext;
+            context.initNow();
         }
 
         @Override
-        public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) {
-            executionContext.initNow();
-            context = executionContext;
+        public long getTimestamp(Record rec) {
+            return context.getNow();
         }
 
         @Override
-        public boolean isRuntimeConstant() {
+        public boolean isConstant() {
```

Comment:
Concept of `Runtime Constant` was invented specifically for `now()` function. Very surprised to see it's reversed

Reply:
yes, indeed, I have reverted this.